cohttp-eio/src/parser.ml

```diff
@@ -0,0 +1,228 @@
+open Angstrom
```

Comment:
The new parser in the Http module is pretty fast, is there a reason you needed to reimplement it?

Reply:
If bigstrings will make it easier for `eio` to consume the parser from the Http module, we can revisit the choice of using bytes there, and update everything to work with big strings. Lwt/Async shouldn't have any issues working with big strings.